UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -383,33 +325,15 @@ public void RequestAction()
             m_RequestAction = true;
         }
 
-        /// <summary>
-        /// Indicates if the agent has reached his maximum number of steps.
-        /// </summary>
-        /// <returns>
-        /// <c>true</c>, if max step reached was reached, <c>false</c> otherwise.
-        /// </returns>
-        public bool IsMaxStepReached()
```

Comment:
These seem like they would be useful for users to have access to (although maybe reworking them to properties).

Reply:
I don't see a use case where the user knowing when the Agent is Done or reached max step would be useful. Do you have an example ?

Reply:
I do not have an example, but I'm sure some users will after we remove it, and it seems like very little overhead to keep.

Reply:
IsMaxStepReached can be derived from GetStepCount and maxStep, so I that might be OK to remove. But I think keeping it as a method hides the internals better (i.e. you have to know about maxStep <= 0  ==> IsMaxStepReached is never true)

Reply:
For Done(), it's generally really frustrating to be able to set something but not read it's value.

Reply:
> I don't see a use case where the user knowing when the Agent is Done or reached max step would be useful. Do you have an example?

@vincentpierre I don't think the argument "I can't think of a reason to have it" is sufficient to remove the accessor of a state that is directly modifiable by the user.  We use it internally in Agent for a number of reasons.  One of the thousands of users that we have may use the `IsDone` accessor for reasons we haven't thought of.  Whether it's custom UI, an event trigger for something else.  I'm not sure how they might use it, but it feels really strange to me to let users modify the state of Agent without being to ask what that state is.  

I would argue that, in general, if we are allowing a user to modify a state, we should also allow them to access that state.

Reply:
A concrete example of why we should be able to read it is right in our own tests:
https://github.com/Unity-Technologies/ml-agents/blob/106e5b453569631da99451eb66de809e1af90ab2/UnitySDK/Assets/ML-Agents/Editor/Tests/MLAgentsEditModeTest.cs#L11-L26

Reply:
a concrete of why we should be able to read it is right in our own tests:
https://github.com/Unity-Technologies/ml-agents/blob/106e5b453569631da99451eb66de809e1af90ab2/UnitySDK/Assets/ML-Agents/Editor/Tests/MLAgentsEditModeTest.cs#L23-L26

Reply:
Yes, we are testing the inner working of the Agent in the tests. I think it is okay, I suppose m_Done can be marked as an internal field on the Agent rather than private.
I think what we need to have access to for testing and for using the tool are very different things.